api/v1alpha1/marin3r_types.go

```diff
@@ -0,0 +1,272 @@
+	DefaultHostForHttp10 *string `json:"defaultHostForHttp10,omitempty"`
+	// Enable http2 in the listener.Disabled by default.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:default:=false
+	// +optional
+	EnableHttp2 *bool `json:"enableHttp2,omitempty"`
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:default:=true
+	// +optional
+	// Allow headers with underscores
+	AllowHeadersWithUnderscores *bool `json:"allowHeadersWithUnderscores,omitempty"`
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	// Max connection duration. If unset no max connection duration will be applied.
+	MaxConnectionDuration *metav1.Duration `json:"maxConnectionDuration,omitempty"`
+}
+
+// RateLimitOptions contains options for the ratelimit filter of the
+// http connection manager
+type RateLimitOptions struct {
+	// The rate limit domain
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	Domain string `json:"domain"`
+	// Whether to allow requests or not if the rate limit service
+	// is unavailable
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:default:=false
+	// +optional
+	FailureModeDeny *bool `json:"failureModeDeny,omitempty"`
+	// Max time to wait for a response from the rate limit service
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:validation:Format:=duration
+	Timeout metav1.Duration `json:"timeout"`
+	// Location of the rate limit service. Must point to one of the
+	// defined clusters.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	RateLimitCluster string `json:"rateLimitCluster"`
+}
+
+// Cluster contains options for an Envoy cluster protobuffer message
+type Cluster struct {
+	EnvoyDynamicConfigMeta `json:",inline"`
+	EnvoyDynamicConfigRaw  `json:",inline"`
+	// The upstream host
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	Host string `json:"host"`
+	// The upstream port
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	Port uint32 `json:"port"`
+	// Specifies if the upstream cluster is http2 or not (default).
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +kubebuilder:default:=false
+	// +optional
+	IsHttp2 *bool `json:"isHttp2"`
+}
+
+// RouteConfiguration contains options for an Envoy route_configuration
+// protobuffer message
+type RouteConfiguration struct {
+	EnvoyDynamicConfigMeta `json:",inline"`
+	EnvoyDynamicConfigRaw  `json:",inline"`
+	// The virtual_hosts definitions for this route configuration.
+	// Virtual hosts must be specified using directly Envoy's API
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	VirtualHosts []runtime.RawExtension `json:"virtualHosts"`
+}
+
+// Runtime contains options for an Envoy runtime protobuffer message
+type Runtime struct {
+	EnvoyDynamicConfigMeta `json:",inline"`
+	EnvoyDynamicConfigRaw  `json:",inline"`
+	// The list of listeners to apply overload protection limits to
```

Comment:
Is this decription complete?

Reply:
yes... why?

Reply:
nothing, I thought it might be incomplete, now I've checked about runtime spec and have understood it